iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -572,22 +647,24 @@ Viewer.prototype.highlightAllTags = function (on, namespaceGroups) {
     var report = this._report;
     var viewer = this;
     if (on) {
-        $(".ixbrl-element:not(.ixbrl-continuation)", this._contents).each(function () {
-            var factId = $(this).data('ivid')[0];
-            var ixn = viewer._ixNodeMap[factId];
-            var elements = viewer.elementsForItemIds([factId].concat(ixn.continuationIds()));
-            elements.addClass("ixbrl-highlight");
-
-            if (!ixn.footnote) {
-                var i = groups[report.getItemById(factId).conceptQName().prefix];
-                if (i !== undefined) {
-                    elements.addClass("ixbrl-highlight-" + i);
+        $(".ixbrl-element", this._contents).each(function () {
+            const factId = $(this).data('ivid')[0];
+            const ixn = viewer._ixNodeMap[factId];
+            if (!ixn.isContinuation) {
```

Comment:
Ah - it wasn't intended to be the same logic, but it was broken.  The same element could be both a "primary" iXBRL element, and a continuation, so we shouldn't filter out all elements that have the `ixbrl-continuation` class - although it would still have been highlighted as the primary element for the continuation also highlights its continuations (unless that primary element was also sharing a table cell with a continuation...)

The updated code uses the first non-continuation element to determine the color of the highlighting which is still an arbitrary choice, but I think preferable to basing it on a continuation.

I think we need to revisit the highlighting of nested tags anyway (see #326).

Reply:
The new logic makes much more sense to me, thanks.